src/utils/raster_utils.py

```diff
@@ -250,31 +267,36 @@ def upsample_raster(ds, resampled_resolution=UPSAMPLED_RESOLUTION, logger=None):
         )
         ds = ds.rio.write_crs("EPSG:4326")
 
-    # Forecast data will have 4 dims, since we have a leadtime
-    nd = len(list(ds.dims))
-    if nd == 4:
+    if fourth_dim:  # 4D case
         resampled_arrays = []
-        for lt in ds.leadtime.values:
-            ds_ = ds.sel(leadtime=lt)
+        
+        for val in ds[fourth_dim].values:
+            ds_ = ds.sel({fourth_dim: val})
             ds_ = ds_.rio.reproject(
                 ds_.rio.crs,
                 shape=(new_height, new_width),
                 resampling=Resampling.nearest,
                 nodata=np.nan,
             )
-            ds_ = ds_.expand_dims(["leadtime"])
+            # Expand along the fourth dimension
+            if fourth_dim == "band":
```

Comment:
Add todo here so we can improve the logic around this for future datasets with 'band'